### ISSUE TITLE
Internally dropping use of separate alpha value

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#7430]
Having a separate alpha attribute to artists is problematic, mostly because colors can carry their own alpha values, and we then have priority issues (e.g. #7375).  I think a much better approach would be to internally only store colors with alpha values, and let `set_alpha` and friends directly change this alpha value.

**Comment by tacaswell** [comment]
It may have to be handled on an artist-by-artist case.  For `AxesImage` it is very nice to have an artist level alpha independent of the alpha in the color map.

**Comment by anntzer** [comment]
Could the alpha value not simply be overwritten to (a copy of) the colormap?

**Comment by tacaswell** [comment]
But you would like to persist the alpha value if you change the color map.

**Comment by anntzer** [comment]
Good point.

**Comment by anntzer** [comment]
Perhaps a reasonable compromise would be to maintain a separate alpha channel only for ScalarMappables (which don't really have an internal notion of own color).

**Comment by tacaswell** [comment]
Probably collections as well still need to carry their own alpha (which maybe should be applied on top of any individual color alpha?) as well.

**Comment by stippingerm** [comment]
This is an interesting implementation question. The good user experience with graphics software tells me that instead of prioritizing one or another alpha value, one should use their combination as follows:
* a global, `collection_level_alpha` value < 1 is a request for the collection being transparent to a certain level
* in addition, individual RGBA values with A<1 indicate the some elements of the collection shall be made more transparent (than the most solid color allowed by `collection_level_alpha`)

In summary, I suggest to **take the product** of `collection_level_alpha` and A from RGBA when displaying. I wrote about that in a [different context](#13035) and that's what happens in Inkscape/Gimp when you set a transparency value on a layer that already contains RGBA colors.

**Comment by anntzer** [comment]
your link looks wrong

**Comment by anntzer** [comment]
I guess this would still be conceptually nice, but I don't really have the bandwidth to push this forward, so we may close this if no one else is interested.

**Comment by timhoffm** [comment]
The reasonable way forward would be using an internal Color class (and additionally ColorArray). This can abstract away all color handling aspects. - For example also the topic of (non-)normalization (what color representation do we return - the one that was provided or some standard (rgb(a)).

However this is not a priority. While not quite well structured, colors work reasonably well in practice.

**Comment by QuLogic** [comment]
@ksunden has added typing for, at least, the input of colours. While that does not specifically limit what gets used internally, it may inform the design of things here a bit.
